DataViews/Program.cs

```diff
@@ -371,6 +375,74 @@ await foreach (var value in values)
                 }
 
                 #endregion //step11
+
+                // Step 12 - Add Units of Measure Column
+                #region step12
+                Console.WriteLine("Step 12: Add Units of Measure Column");
+                fieldSet = dataView.DataFieldSets.Single(a => a.QueryId == sampleQueryId);
+
+                var uomField1 = fieldSet.DataFields.Single(a => a.Keys.Contains(uomColumn1));
+                var uomField2 = fieldSet.DataFields.Single(a => a.Keys.Contains(uomColumn2));
+
+                uomField1.IncludeUom = true;
+                uomField2.IncludeUom = true;
+
+                await dataviewService.CreateOrUpdateDataViewAsync(dataView).ConfigureAwait(false);
+
+                values = dataviewService.GetDataInterpolatedAsync(
+                    dataView.Id,
+                    OutputFormat.Default,
+                    sampleStartTime.ToUniversalTime().ToString(CultureInfo.InvariantCulture),
+                    sampleEndTime.ToUniversalTime().ToString(CultureInfo.InvariantCulture),
+                    sampleInterval.ToString(),
+                    null,
+                    CacheBehavior.Refresh,
+                    default);
+
+                await foreach (var value in values)
+                {
+                    Console.WriteLine(value);
+                }
+                #endregion //step 12
+
+                // Step 13 - Add Summary Columns
+                #region step13
+                Console.WriteLine("Step 13: Add Summaries Columns");
+                fieldSet = dataView.DataFieldSets.Single(a => a.QueryId == sampleQueryId);
+
+                var fieldToSummarize = fieldSet.DataFields.Single(a => a.Keys.Contains(summaryField));
+
+                // Make two copies of the field to be summarized
+                var summaryField1 = fieldToSummarize.Clone();
+                var summaryField2 = fieldToSummarize.Clone();
+
+                // Set the summary properties on the new fields and add them to the FieldSet
+                summaryField1.SummaryDirection = SummaryDirection.Forward;
+                summaryField1.SummaryType = summaryType1;
+
+                summaryField2.SummaryDirection = SummaryDirection.Forward;
+                summaryField2.SummaryType = summaryType2;
+
+                fieldSet.DataFields.Add(summaryField1);
+                fieldSet.DataFields.Add(summaryField2);
+
+                await dataviewService.CreateOrUpdateDataViewAsync(dataView).ConfigureAwait(false);
+
+                values = dataviewService.GetDataInterpolatedAsync(
+                    dataView.Id,
+                    OutputFormat.Default,
+                    sampleStartTime.ToUniversalTime().ToString(CultureInfo.InvariantCulture),
```

Comment:
would there be any benefit to a helper for this?  definitely not a huge deal

Reply:
yea this did get a little ridiculous by the end with the addition of two more CRUD operations that each output the dataview data. I should add a helper function at this point